src/main/java/com/bgsoftware/superiorskyblock/island/GridManagerImpl.java

```diff
@@ -356,7 +356,7 @@ private void createIslandInternalOnSuccessCallback(IslandBuilderImpl builder, Bi
             if (updateGamemode)
                 player.setGameMode(GameMode.SURVIVAL);
 
-            if (!teleportPlayer) {
+            if (!teleportPlayer || !plugin.getSettings().isTeleportOnCreate()) {
```

Comment:
Events should be able to override this feature.
The `isTeleportOnCreate` should be forwarded to the `IslandCreateEvent`, and then this check should be reverted